src/utils.ts

```diff
@@ -1,39 +1,57 @@
-const setPaddingBetweenNodes = ({
-  context,
-  problemNode,
-  firstNode,
-  message,
-}) => {
-  context.report({
-    node: problemNode,
-    message,
-    fix: function(fixer) {
-      return fixer.insertTextAfter(firstNode, '\n');
-    },
-  });
-};
+import padding from './rules/padding';
 
-const getLeftSibling = node => {
-  const siblings = node.parent.body;
-  const nodePosition = siblings.indexOf(node);
-  return siblings[nodePosition - 1];
-};
+type TokenIdentifier =
+  | '*'
+  | 'afterAll'
+  | 'afterEach'
+  | 'beforeAll'
+  | 'beforeEach'
+  | 'describe'
+  | 'expect'
+  | 'it'
+  | 'test';
+
+interface Option {
+  blankLine: 'always' | 'any' | 'never';
+  prev: TokenIdentifier | TokenIdentifier[];
+  next: TokenIdentifier | TokenIdentifier[];
+}
 
-const getStartLine = node => node && node.loc.start.line;
+interface RuleDefs {
+  [name: string]: Option[];
+}
 
-const getEndLine = node => node && node.loc.end.line;
+/**
+ * Extend the base padding rule to make a new rule
+ */
+export const makeRule = (options: Option[]) => {
+  return {
+    meta: {
+      type: 'layout',
+      fixable: 'whitespace',
+    },
+    // Wrap base padding create with a function that modifies RuleContext
+    // with the options for the new rule
+    create(context) {
```

Comment:
Can we have a type definition for this? I believe this is context? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/eslint/index.d.ts#L297

Reply:
Yep, it's exactly that.